src/screens/Profile/Notification.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { View, StyleSheet, AsyncStorage } from 'react-native';
+import { Card, Button, Paragraph } from 'react-native-paper';
+import auth from '@react-native-firebase/auth';
+import Swipeable from 'react-native-gesture-handler/Swipeable';
+import PropTypes from 'prop-types';
+
+export default class Notification extends React.Component {
+  constructor(props) {
+    super(props);
+    const { viewed } = this.props;
+    this.state = { viewed, visible: true };
+    this.onClearPress = this.onClearPress.bind(this);
+    this.onViewPress = this.onViewPress.bind(this);
+  }
+
+  componentDidMount() {
+    this.unsubscribeFromAuth = auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({ currentUserID: user.uid });
+      }
+    });
+  }
+
+  async onViewPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then((value) => {
+      const notificationArray = JSON.parse(value);
+      const notif = notificationArray.find((el) => el.id === notificationID);
+      if (notif.viewed === false) {
+        const updatedArray = notificationArray.filter((val) => val.id !== notificationID);
+        notif.viewed = true;
+        updatedArray.push(notif);
```

Comment:
This is can be made more efficient. I think if you set `notif.viewed = true`, the `notificationArray` will get modified. But if I'm wrong, you can also use `notificationArray.findIndex()` to find where the the item you want is, then use that index to modify the array directly. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex